server/app/initializers/01_symmetric_encryption.rb

```diff
@@ -1,10 +1,39 @@
 require 'symmetric-encryption'
 require 'symmetric_encryption/extensions/mongoid/encrypted'
+require 'hkdf'
 
-if ENV['VAULT_KEY'] && ENV['VAULT_IV']
+if vault_key = ENV['VAULT_KEY']
+  # generate strong keys using entropy derived from the weak (human-readable) input key
+  hkdf = HKDF.new(vault_key)
+
+  # primary cipher used for encryption
+  # not configured using any fixed IV, assuming that we always encrypt with a random IV
+  # default IV implies an IV of all zeroes, in case something does encrypt without random_iv: true
```

Comment:
> test what happens for SymmetricEncryption.encrypt(..., random_iv: false) without configuring any SymmetricEncryption::Cipher.new(iv: ...)

You end up with `iv: nil`, which ends up using the OpenSSL default of an all-zeroes IV:

```
irb(main):024:0> SymmetricEncryption::Cipher.parse_header!(SymmetricEncryption.cipher.decode(SymmetricEncryption.encrypt('foobar', false)))
=> #<struct SymmetricEncryption::Cipher::HeaderStruct compressed=false, iv=nil, key=nil, cipher_name=nil, version=1, decryption_cipher=#<SymmetricEncryption::Cipher:0x2adcedbbd6b4 @key="[FILTERED]" @iv=nil @cipher_name="aes-256-cbc", @version=1, @encoding=:base64, @always_add_header=true>
```

From: https://www.reddit.com/r/ruby/comments/ndl1p/be_careful_rubys_openssl_rulez/

> Currently the "default IV" being used is an all zeroes IV (see https://gist.github.com/1480700), which is also not ideal, but note that this default is actually OpenSSL's default and it needs to be kept for the same reason of backwards compatibility.

See https://gist.github.com/emboss/1480700